src/sqlfluff/testing/rules.py

```diff
@@ -142,3 +142,7 @@ def rules__test_helper(test_case):
         # If a `fixed` value is provided then check it matches
         if test_case.fix_str:
             assert res == test_case.fix_str
+        else:
+            # Check that tests without a fix_str do not apply any fixes.
+            # If the fix is expected, add the missing fix_str in the test case.
+            assert res == test_case.fail_str
```

Comment:
I wonder if we need an assertion error here to make it more obvious why we are asserting this?

Something like “Please supply a fix_str if the rule can apply a fix for this test”. Or something like that?

Reply:
Or "No fix_str was provided, but the rule modified the SQL. Please add a fix_str to the test."

Reply:
I committed the suggestion by @tunetheweb in https://github.com/sqlfluff/sqlfluff/pull/2663/commits/311480392f4e99f0d1542d8d93d4a0389176b6f2